### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by SiddhiRao10**
wdio and python code files pushed

### CODE REVIEW

File: nodejs-wdio/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+  This run profile executes a specific test scenario on a single browser instance on your own machine.
+
+
+
+
+
+## Running Your Tests
+
+### Run a specific test on BrowserStack
```
Comment (nithyamn) [inline nodejs-wdio/README.md:59]: Lets remove this part for running on bstack as we would want them to integrate this

File: nodejs-wdio/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@axe-core/webdriverio": "^4.3.1",
+    "@rpii/wdio-html-reporter": "^6.3.1",
+    "@types/node": "^20.8.6",
+    "@wdio/allure-reporter": "^8.17.0",
+    "@wdio/browserstack-service": "^8.1.3",
```
Comment (nithyamn) [inline nodejs-wdio/package.json:17]: We can remove the browserstack specific services

File: nodejs-wdio/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    "@wdio/cli": "^8.17.0",
+    "@wdio/globals": "^8.1.3",
+    "@wdio/junit-reporter": "^8.1.2",
+    "@wdio/local-runner": "^8.1.3",
+    "@wdio/mocha-framework": "^8.1.2",
+    "@wdio/spec-reporter": "^8.1.2",
+    "@wdio/sync": "^6.11.0",
+    "allure-commandline": "^2.13.8",
+    "browserstack-local": "1.4.8",
```
Comment (nithyamn) [inline nodejs-wdio/package.json:26]: We can remove the browserstack specific services

File: python/browserstack.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# =============================
+# Set BrowserStack Credentials
```
Comment (nithyamn) [inline python/browserstack.yml:2]: Lets remove this file as we want the users to integrate this part

File: python/conftest.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+TASK_ID = int(os.environ['TASK_ID']) if 'TASK_ID' in os.environ else 0
+
+
+with open(CONFIG_FILE) as data_file:
+    CONFIG = json.load(data_file)
+
+bs_local = None
+timenow = None
+
+BROWSERSTACK_USERNAME = os.environ['BROWSERSTACK_USERNAME'] if 'BROWSERSTACK_USERNAME' in os.environ else CONFIG["user"]
```
Comment (nithyamn) [inline python/conftest.py:20]: Lets remove browserstack related code segments and keep it in vanilla format

File: python/pavement.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    packages=['tests']
+)
+
+def run_py_test(config, run_type, task_id=0):
+    if config == 'local' and run_type == 'remote':
+        if platform.system() == "Windows":
+            sh('cmd /C "set CONFIG_FILE=resources/%s.json && set TASK_ID=%s && set REMOTE=true && pytest -s src/test/suites/e2e.py --driver Browserstack -n 1"' % (config, task_id))
+        else:
+            sh('CONFIG_FILE=resources/%s.json TASK_ID=%s REMOTE=true pytest -s src/test/suites/e2e.py --driver Browserstack -n 1' % (config, task_id))
+    elif run_type == 'remote':
```
Comment (nithyamn) [inline python/pavement.py:25]: Lets remove browserstack related code segments and keep it in vanilla format

File: python/src/test/suites/e2e.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    order_confirm.wait_for_confirmation_message()
+    if os.environ['REMOTE'] == "true":
+        order_confirm.click_download_pdf()
+        order_confirm.download_exists(driver, 'confirmation.pdf' )
+    order_confirm.click_continue_shopping()
+    home.navigate_to_orders()
+    orders = OrdersPage(driver)
+    status = orders.verify_orders_placed()
+    time.sleep(5)
+    if os.environ['REMOTE'] == "true":
```
Comment (nithyamn) [inline python/src/test/suites/e2e.py:45]: Lets remove browserstack related code segments and keep it in vanilla format

File: python/src/test/suites/login.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+            driver.execute_script('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "Username not found. Login Failed"}}')
+
+
+@pytest.mark.nondestructive
+def test_login_locked_user(driver, base_url):
+    login = LoginPage(driver)
+    login.open_base_url(base_url)
+    login.sign_in("locked_user","testingisfun99")
+    text = login.wait_element_present((By.CLASS_NAME, 'api-error')).text
+    if os.environ['REMOTE'] == "true":
```
Comment (nithyamn) [inline python/src/test/suites/login.py:34]: Lets remove browserstack related code segments and keep it in vanilla format

File: python/src/test/suites/offers.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    WebDriverWait(driver, 20).until(EC.visibility_of_element_located((By.CSS_SELECTOR, '#password input')))
+    password = driver.find_element(By.CSS_SELECTOR, "#password input")
+    password.send_keys("testingisfun99\n")
+    #Clicking on Login
+    WebDriverWait(driver, 20).until(EC.element_to_be_clickable((By.ID, 'login-btn')))
+    driver.find_element(By.ID, "login-btn").click()
+    #Click on Offers
+    WebDriverWait(driver, 20).until(EC.element_to_be_clickable((By.ID, 'offers')))
+    driver.find_element(By.ID, "offers").click()
+    if os.environ['REMOTE'] == "true":
```
Comment (nithyamn) [inline python/src/test/suites/offers.py:32]: Lets remove browserstack related code segments and keep it in vanilla format

File: python/src/test/suites/product.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+@pytest.mark.nondestructive
+def test_price_filter(driver, base_url):
+    login = LoginPage(driver)
+    login.open_base_url(base_url)
+    login.sign_in("fav_user","testingisfun99")
+    home = HomePage(driver)
+    home.select_lowest_price_filter()
+    time.sleep(5)
+    if os.environ['REMOTE'] == "true":
```
Comment (nithyamn) [inline python/src/test/suites/product.py:37]: Lets remove browserstack related code segments and keep it in vanilla format

File: python/src/test/suites/user.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        else:
+            driver.execute_script('browserstack_executor: {"action": "setSessionStatus", "arguments": {"status":"failed","reason": "No Existing Orders"}}')
+
+@pytest.mark.nondestructive
+def test_no_image(driver, base_url):
+    login = LoginPage(driver)
+    login.open_base_url(base_url)
+    login.sign_in("image_not_loading_user","testingisfun99")
+    home = HomePage(driver)
+    if os.environ['REMOTE'] == "true":
```
Comment (nithyamn) [inline python/src/test/suites/user.py:36]: Lets remove browserstack related code segments and keep it in vanilla format

File: nodejs-wdio/README.md (hunk 1 above)
Comment (SiddhiRao10) [reply nodejs-wdio/README.md]: Updated as per suggestion.

File: nodejs-wdio/package.json (hunk 3 above)
Comment (SiddhiRao10) [reply nodejs-wdio/package.json]: Updated as per suggestion.

File: python/browserstack.yml (hunk 4 above)
Comment (SiddhiRao10) [reply python/browserstack.yml]: Updated as per suggestion.

File: python/conftest.py (hunk 5 above)
Comment (SiddhiRao10) [reply python/conftest.py]: Updated as per suggestion.

File: python/pavement.py (hunk 6 above)
Comment (SiddhiRao10) [reply python/pavement.py]: Updated as per suggestion.

File: python/src/test/suites/e2e.py (hunk 7 above)
Comment (SiddhiRao10) [reply python/src/test/suites/e2e.py]: Updated as per suggestion.

File: python/src/test/suites/product.py (hunk 10 above)
Comment (SiddhiRao10) [reply python/src/test/suites/product.py]: Updated as per suggestion.

File: python/src/test/suites/user.py (hunk 11 above)
Comment (SiddhiRao10) [reply python/src/test/suites/user.py]: Updated as per suggestion.

**Review by nithyamn — APPROVED** [review]

**Comment by SiddhiRao10** [comment]
Closing this , and creating new PR to merger from develop to develop

**Pull request closed without merging**